src/core/server/saved_objects/saved_objects_type_registry.mock.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { SavedObjectTypeRegistry } from './saved_objects_type_registry';
+
+type RegistryMock = jest.Mocked<SavedObjectTypeRegistry>;
+
+const createRegistryMock = (): RegistryMock => {
+  const mock = {
+    registerType: jest.fn(),
+    getType: jest.fn(),
+    getAllTypes: jest.fn(),
+    isNamespaceAgnostic: jest.fn(),
+    isHidden: jest.fn(),
+    getIndex: jest.fn(),
+  };
+
+  mock.getIndex.mockReturnValue('.kibana-test');
+  mock.isHidden.mockReturnValue(false);
+  mock.isNamespaceAgnostic.mockImplementation((type: string) => type === 'global');
+
+  return mock as any;
```

Comment:
🤷‍♂ we've done in this in other places. You don't necessarily have to actually export the interface, just use it for the mock and export the mock function. It will be type compatible while still making sure the mock is the correct type.

Reply:
Issue is not about returning the interface, it's that my mock doesn't respect the `jest.Mocked<SavedObjectTypeRegistry>` type, as it's does not have the `readonly types` property. Changing the mock type to ` jest.Mocked<PublicMethodsOf<SavedObjectTypeRegistry>>` fails when using the mock with

```
Argument of type 'Mocked<Pick<SavedObjectTypeRegistry, "getType" | "registerType" | "getAllTypes" | "isNamespaceAgnostic" | "isHidden" | "getIndex">>' is not assignable to parameter of type 'SavedObjectTypeRegistry'.
```

Reply:
The only problem here is that internal mocks aren't type-safe and might be out of sync when `SavedObjectTypeRegistry` updates the interface. https://github.com/elastic/kibana/issues/54906 postponed, probably adding a separate interface is not such a bad idea.

Reply:
Added the `ISavedObjectTypeRegistry` interface